packages/cli/src/commands/publish/publishPackages.ts

```diff
@@ -103,7 +104,10 @@ async function publishAPackage(
     `Publishing ${chalk.cyan(`"${name}"`)} at ${chalk.green(`"${version}"`)}`
   );
 
-  const publishDir = pkg.dir;
+  const publishDir =
+    publishConfig && publishConfig.directory
+      ? join(pkg.dir, publishConfig.directory)
+      : pkg.dir;
```

Comment:
nit: If you rebase on top of master then you should be able to use optional chaining here 🎉 
```suggestion
  const publishDir =
    publishConfig?.directory
      ? join(pkg.dir, publishConfig.directory)
      : pkg.dir;
```